src/main/java/com/laytonsmith/core/functions/StringHandling.java

```diff
@@ -845,6 +846,82 @@ public Construct exec(Target t, Environment env, Construct... args) throws Cance
 	}
 
 	@api
+	public static class char_is_uppercase extends AbstractFunction implements Optimizable {
+
+		@Override
+		public Class<? extends CREThrowable>[] thrown() {
+			return new Class[]{CREFormatException.class, CRECastException.class};
+		}
+
+		@Override
+		public String getName() {
+			return "char_is_uppercase";
+		}
+
+		@Override
+		public Construct exec(Target t, Environment environment, Construct... args) throws CancelCommandException, ConfigRuntimeException {
+			if(!(args[0].nval() instanceof String)) {
+			    throw new CREFormatException(this.getName() + " expects a string as first argument, but type " +
+				    args[0].typeof() + " was found.", t);
+			}
+			String text = args[0].nval();
+			// Enforce the fact we are only taking the first character here
+			// Do not let the user pass an entire string. Only a single character.
+			if(text.length() > 1) {
+			    throw new CRECastException("Got \"" + text + "\" instead of expected character.", t);
```

Comment:
A CREFormatException would be more appropriate here, as well as the exception below. CRECastException is reserved for when, say, an int is expected, but a string is given. In this case, a string would still be given, it would just have the wrong format. See: http://wiki.sk89q.com/wiki/CommandHelper/Staged/Exceptions#FormatException


The exception above should be a CastException though.